From 1c4896f15f63bb6719ece59b626fddf55b766a0e Mon Sep 17 00:00:00 2001
From: Julio Galvan <julio-void@epazote.net>
Date: Sat, 19 Sep 2020 21:32:19 -0700
Subject: [PATCH] Support meson

I am trying to package this library for voidlinux but it fails to
crossbuild automatically. I thought that we could use this for now.
---
 meson.build | 42 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)
 create mode 100644 meson.build

diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..8b6c90e
--- /dev/null
+++ meson.build
@@ -0,0 +1,42 @@
+project('purple-mm-sms', 'c',
+  version: '0.1.7',
+)
+
+cc = meson.get_compiler('c')
+
+purple = dependency('purple')
+mm = dependency('mm-glib')
+glib= dependency('glib-2.0', version: '>=2.50.0')
+
+purple_plugdir = purple.get_pkgconfig_variable('plugindir')
+
+mm_sms_headers = [
+  'mm-sms.h',
+  'itu-e212-iso.h',
+]
+
+mm_sms_sources = [
+  mm_sms_headers,
+  'mm-sms.c',
+]
+
+mm_sms_deps = [
+  purple,
+  mm,
+  glib,
+]
+
+mm_sms_shared = shared_library(
+  'mm-sms',
+  mm_sms_sources,
+         dependencies : mm_sms_deps,
+         name_prefix  : '',
+          install_dir : purple_plugdir,
+              install : true,
+)
+
+
+iconsdir = join_paths(get_option('datadir'), 'pixmaps', 'pidgin', 'protocols')
+foreach size: ['16', '22', '48']
+  install_data('icons/mm-sms-' + size + 'px.png', rename: 'mm-sms.png', install_dir: iconsdir + '/' + size)
+endforeach
-- 
2.28.0

